backend/src/email/services/email-template.service.ts

```diff
@@ -234,16 +235,41 @@ const addToMessageLogs = async (
   }
 }
 
+/**
+ * Checks for invalid email recipients from an array of records
+ *
+ * @param records
+ */
 const hasInvalidEmailRecipient = (
   records: MessageBulkInsertInterface[]
 ): boolean => {
   return records.some((record) => !validator.isEmail(record.recipient))
 }
 
+/**
+ * Attempts to hydrate the first record.
+ * @param records
+ * @param templateBody
+ * @param templateSubject - optional
+ */
+const testHydration = (
```

Comment:
I could abstract `testHydration` into `TemplateService` in `core`, but I find that it is cleaner this way, since SMS doesn't have `templateSubject`.